libs/designer-ui/src/lib/floatingactionmenu/floatingactionmenubase/index.tsx

```diff
@@ -0,0 +1,147 @@
+import type { DynamicallyAddedParameterTypeType } from '../../dynamicallyaddedparameter';
+import { getMenuItemsForDynamicAddedParameters } from './helper';
+import { KeyCodes } from '@fluentui/react';
+import { useBoolean } from '@fluentui/react-hooks';
+import { ValidationErrorCode, ValidationException } from '@microsoft/utils-logic-apps';
+import type { PropsWithChildren } from 'react';
+import React from 'react';
+import { useIntl } from 'react-intl';
+
+export type FloatingActionMenuItem = {
+  type: DynamicallyAddedParameterTypeType;
+  icon: string;
+  label: string;
+};
+
+type FloatingActionMenuBaseProps = {
+  supportedTypes: string[];
+  collapsedTitle: string;
+  expandedTitle: string;
+  onMenuItemSelected: (item: FloatingActionMenuItem) => void;
+};
+
+export const FloatingActionMenuBase = (props: PropsWithChildren<FloatingActionMenuBaseProps>): JSX.Element => {
+  const intl = useIntl();
+  const [expanded, { toggle: toggleExpanded }] = useBoolean(false);
+
+  if (!props.supportedTypes?.length) {
+    throw new ValidationException(ValidationErrorCode.INVALID_PARAMETERS, 'supportedTypes are necessary.');
+  }
+
+  const menuItems = getMenuItemsForDynamicAddedParameters(props.supportedTypes);
+
+  const toggleExpandedOnKeyDown = (e: React.KeyboardEvent<HTMLElement>): void => {
+    const { keyCode } = e;
+
+    if (keyCode === KeyCodes.enter || keyCode === KeyCodes.space) {
+      e.preventDefault();
+      e.stopPropagation();
+      toggleExpanded();
+    }
+  };
+
+  const renderMenuButton = (): JSX.Element => {
+    return (
+      <div role="button" className="msla-floating-action-menu" onClick={toggleExpanded} onKeyDown={toggleExpandedOnKeyDown} tabIndex={0}>
+        <span className="msla-floating-action-menu-plus-icon">{'+'}</span>
```

Comment:
Do we have actual Fluent icons we can use here?

Reply:
swapped to icons